crates/matrix-sdk-ui/src/timeline/event_item/mod.rs

```diff
@@ -1150,24 +1148,17 @@ mod tests {
         display_name: &str,
         avatar_url: &str,
     ) -> Raw<AnySyncTimelineEvent> {
-        sync_timeline_event!({
-            "type": "m.room.member",
-            "content": {
-                "avatar_url": avatar_url,
-                "displayname": display_name,
-                "membership": "join",
-                "reason": ""
-            },
-            "event_id": "$143273582443PhrSn:example.org",
-            "origin_server_ts": 143273583,
-            "room_id": room_id,
-            "sender": "@example:example.org",
-            "state_key": user_id,
-            "type": "m.room.member",
-            "unsigned": {
-              "age": 1234
-            }
-        })
+        EventFactory::new()
+            .room(room_id)
+            .sender(user_id!("@example:example.org"))
+            .member(user_id)
```

Comment:
That's the correct order, fwiw: the `user_id` passed to `member()` ends up being the state_key (and the sender, when it's not set, but here it's set so we're all good!). Good job!

Reply:
Thanks for the feedback, this was also how I had initially understood the interplay between the two methods, so I'm glad I was following things correctly :slightly_smiling_face: